hw/i386/virt/memory.c

```diff
@@ -25,7 +25,7 @@
 #include "qemu/error-report.h"
 #include "cpu.h"
 
-#define VIRT_LOWMEM 0x80000000
+#define VIRT_LOWMEM 0x40000000
```

Comment:
@liujing2 if you change this, you need to make a corresponding change in OVMF where the value is hardcoded in a GUID.

Reply:
gEfiMdePkgTokenSpaceGuid.PcdPciExpressBaseAddress

Reply:
@liujing2 additionally OVMF ignores the E820 reservations. Hence we have to send in the number of segments also via fw_cfg. This will allow OVMF to setup the EFI memory map correctly so that each segment's range is added properly.